src/hyperlight_host/src/hypervisor/hyperv_windows.rs

```diff
@@ -482,158 +460,53 @@ impl Hypervisor for HypervWindowsDriver {
-
-            self.processor.run()?
-        };
-
-        // Clear running bit
-        interrupt_handle_internal.clear_running_bit();
-
-        let is_canceled = exit_context.ExitReason == WHV_RUN_VP_EXIT_REASON(8193i32); // WHvRunVpExitReasonCanceled
-
-        // Check if this was a manual cancellation (vs internal Windows cancellation)
-        let cancel_was_requested_manually =
-            interrupt_handle_internal.is_cancel_requested_for_generation(generation);
-
-        // Only clear cancel_requested if we're actually processing a cancellation for this generation
-        if is_canceled && cancel_was_requested_manually {
-            interrupt_handle_internal.clear_cancel_requested();
+    #[expect(non_upper_case_globals, reason = "Windows API constant are lower case")]
+    fn run_vcpu(&mut self) -> Result<VmExit> {
+        let mut exit_context: WHV_RUN_VP_EXIT_CONTEXT = Default::default();
+
+        unsafe {
+            WHvRunVirtualProcessor(
+                self.processor.get_partition_hdl(),
+                0,
+                &mut exit_context as *mut _ as *mut c_void,
+                std::mem::size_of::<WHV_RUN_VP_EXIT_CONTEXT>() as u32,
+            )?;
         }
 
-        #[cfg(gdb)]
-        let debug_interrupt = self
-            .interrupt_handle
-            .debug_interrupt
-            .load(Ordering::Relaxed);
-
         let result = match exit_context.ExitReason {
-            // WHvRunVpExitReasonX64IoPortAccess
-            WHV_RUN_VP_EXIT_REASON(2i32) => {
-                // size of current instruction is in lower byte of _bitfield
-                // see https://learn.microsoft.com/en-us/virtualization/api/hypervisor-platform/funcs/whvexitcontextdatatypes)
+            WHvRunVpExitReasonX64IoPortAccess => unsafe {
                 let instruction_length = exit_context.VpContext._bitfield & 0xF;
-                unsafe {
-                    debug!(
-                        "HyperV IO Details :\n Port: {:#x} \n {:#?}",
-                        exit_context.Anonymous.IoPortAccess.PortNumber, &self
-                    );
-                    HyperlightExit::IoOut(
-                        exit_context.Anonymous.IoPortAccess.PortNumber,
-                        exit_context
-                            .Anonymous
-                            .IoPortAccess
-                            .Rax
-                            .to_le_bytes()
-                            .to_vec(),
-                        exit_context.VpContext.Rip,
-                        instruction_length as u64,
-                    )
-                }
-            }
-            // HvRunVpExitReasonX64Halt
-            WHV_RUN_VP_EXIT_REASON(8i32) => {
-                debug!("HyperV Halt Details :\n {:#?}", &self);
-                HyperlightExit::Halt()
-            }
-            // WHvRunVpExitReasonMemoryAccess
-            WHV_RUN_VP_EXIT_REASON(1i32) => {
+                let rip = exit_context.VpContext.Rip + instruction_length as u64;
```

Comment:
On windows do we still need to increment RIP?

Reply:
Yes we are doing it still. It was done before, but in `handle_io`